src/components/Carousel.tsx

```diff
@@ -0,0 +1,39 @@
+import Link from "next/link";
+import { IProduct, ProductItemCard } from "./ProductItemCard";
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+export function Carousel() {
+    const [products, setProducts] = useState<IProduct[]>([]);
+
+    const fetchProducts = async () => {
+        try {
+            const response = await axios.get('http://localhost:3001/products');
+            setProducts(response.data);
+        } catch (err) {
+            console.error('Erro ao carregar os produtos');
+        }
+    };
+
+    useEffect(() => {
+        fetchProducts();
+    }, []);
```

Comment:
Lembra que comentamos que a premissa principal do next é fazer requisições server side? Nesse caso, a chamada só vai ser feito quando o componente for montado, ou seja, pelo client side. Sugiro que dê uma olhada nessa documentação: https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props